src/Nethermind/Nethermind.Baseline/Tree/BaselineTree.cs

```diff
@@ -193,6 +202,32 @@ public BaselineTreeNode[] GetLeaves(params uint[] leafIndexes)
             return leaves;
         }
 
+        public void CalculateHashes(uint startIndex = 0)
+        {
+            var nodesToIterate = Count;
+            uint startingRowIndex = startIndex - startIndex % 2;
+            for (uint row = LeafRow; row > 0; row--)
+            {
+                for (uint indexAtRow = startingRowIndex; indexAtRow < nodesToIterate; indexAtRow += 2)
+                {
+                    Index index = new Index(row, indexAtRow);
+                    var hash = LoadValue(index);
+                    var siblingIndex = index.Sibling();
+                    var siblingHash = LoadValue(siblingIndex);
+                    byte[] parentHash = new byte[32];
+                    Hash(hash.Bytes.AsSpan(), siblingHash.Bytes.AsSpan(), parentHash);
+                    Index parentIndex = index.Parent();
+                    SaveValue(parentIndex, parentHash);
```

Comment:
maybe you can get significant speed increase if you batch insert into the database